agent/eventhandler/task_handler.go

```diff
@@ -221,6 +223,31 @@ func (handler *TaskHandler) taskStateChangesToSend() []api.TaskStateChange {
 
 // batchContainerEventUnsafe collects container state change events for a given task arn
 func (handler *TaskHandler) batchContainerEventUnsafe(event api.ContainerStateChange) {
+	// if this is a managed agent state change, let's try to bundle it in an existing container event
+	if event.GetEventType() == statechange.ManagedAgentEvent {
+		containerChanges, ok := handler.tasksToContainerStates[event.TaskArn]
+		if ok {
+			for i, cs := range containerChanges {
+				if cs.ContainerName == event.ContainerName && cs.Status == event.Status {
+					cs.ManagedAgents = append(cs.ManagedAgents, event.ManagedAgents...)
+					containerChanges[i] = cs
+					return
```

Comment:
just to understand, we will not send a container event with just a managed agent state change if there is an existing container event in queue?

Reply:
That's correct. You can think of it as one more level of batching: managed agent changes are batched in container changes, and container changes is batched in task changes.